test/tests/functional/pbs_power_provisioning_cray.py

```diff
@@ -231,28 +263,128 @@ def test_cray_eoe_job(self):
 
         self.setup_cray_eoe()   # setup hook and eoe
         self.enable_power()		# enable hooks
-
         # Make sure eoe is set correctly on the vnodes
-        eoes = set()		# use sets to be order independent
-        for n in self.server.status(NODE):
-            name = n['id']
-            if 'resources_available.eoe' in n:
-                self.server.manager(MGR_CMD_SET, NODE,
-                                    {"power_provisioning": True}, name)
-                curr = n['resources_available.eoe'].split(',')
-                self.logger.info("%s has eoe values %s" % (name, str(curr)))
-                if len(eoes) == 0:  # empty set
-                    eoes.update(curr)
-                else:  # all vnodes must have same eoes
-                    self.assertTrue(eoes == set(curr))
-        self.assertTrue(len(eoes) > 0)
-
-        # submit jobs for each eoe value
-        x = 5
+        eoes = set(['low', 'med', 'high'])
         while len(eoes) > 0:
             eoe = eoes.pop()
-            jid = self.eoe_job(x, eoe)
+            jid = self.eoe_job(self.npp, eoe)
             self.energy_check(jid)
 
-    def tearDown(self):
-        TestFunctional.tearDown(self)
+    @timeout(700)
+    def test_cray_request_more_eoe(self):
+        """
+        Submit jobs with available+1 eoe chunks and verify job comment.
+        """
+
+        self.setup_cray_eoe()   # setup hook and eoe
+        self.enable_power()             # enable hooks
+        x = self.npp + 1
+        jid = self.submit_job(10,
+                              {'Resource_List.place': 'scatter',
+                               'Resource_List.select': '%d:eoe=%s' % (x,
+                                                                      'high')})
+        self.server.expect(JOB, {
+            'job_state': 'Q',
+            'comment': 'Not Running: No available resources on nodes'},
+            id=jid)
+
+    @timeout(700)
+    def test_cray_eoe_job_multiple_eoe(self):
+        """
+        Submit jobs requesting multiple eoe and job should rejected by qsub.
+        """
+
+        self.setup_cray_eoe()   # setup hook and eoe
+        self.enable_power()             # enable hooks
+
+        a = {'Resource_List.place': 'scatter',
+             'Resource_List.select': '10:eoe=low+10:eoe=high'}
+        j = Job(TEST_USER, attrs=a)
+        j.set_sleep_time(10)
+        try:
+            self.server.submit(j)
+        except PbsSubmitError as e:
+            self.assertTrue(
+                'Invalid provisioning request in chunk' in e.msg[0])
```

Comment:
For negative testing, you need to test both sides.  Right now if the job submits successfully, your test will pass.  I don't think you want this.  say jid = None above the try, set jid to the submit, and do an assertFalse on the jid at the end.